soc/arm/cypress/psoc6/soc.h

```diff
@@ -14,14 +15,17 @@
 #ifndef _SOC__H_
 #define _SOC__H_
 
+#include <sys/util.h>
+
 #ifndef _ASMLANGUAGE
+
+/* Add include for DTS generated information */
+#include <devicetree.h>
+
 #include <cy_device_headers.h>
 
-/* ARM CMSIS definitions must be included before kernel_includes.h.
- * Therefore, it is essential to include kernel_includes.h after including
- * core SOC-specific headers.
- */
-#include <kernel_includes.h>
+#include "../common/soc_gpio.h"
+#include "../common/cypress_psoc6_dt.h"
```

Comment:
Cypress uses PDL with PSoC 4 and PSoC 6. PSoC-6 is an dual core asymmetric (Cortex-M0+/M4) and PSoC-4 is a Cortex-M0+. It will be natural share some code once both shares same library.